### ISSUE TITLE
NVDA does not read the file list in explorer after pasting a new file in the folder.

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#5713]
Dear, another problem I notice on NVDA when using the file explorer on windows 10, is that sometimes after pasting a new file within a folder or after editing the file name NVDA to read the file list within the folder as the folder had lost focus, being necessary to press shift + tabe 
[explorer.txt](https://github.com/nvaccess/nvda/files/104835/explorer.txt)
and tabe to the list of files to be read again.
 To reproduce the error, being within a folder in the file explorer, just paste new files or edit the names of these files, which after some operations NVDA no longer reads the list of files in Windows Explorer.
 Attached a log that shows this fault.

**Comment by Brian1Gaff** [comment]
I've seen this in Windows 7 as well, sometimes explorer crashes in some odd way, often either a restart of explorer works, or if its still running a refresh from the view menu fixes it till the next time.
 

its weird in 7, as it goes off offering to find a solution, but never does.
 Brian

bglists@blueyonder.co.uk 
Sent via blueyonder. 
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
  ----- Original Message ----- 
  From: fernando-jose-silva 
  To: nvaccess/nvda 
  Sent: Tuesday, January 26, 2016 11:51 AM
  Subject: [nvda] NVDA does not read the file list in explorer after pasting a new file in the folder. (#5713)


  Dear, another problem I notice on NVDA when using the file explorer on windows 10, is that sometimes after pasting a new file within a folder or after editing the file name NVDA to read the file list within the folder as the folder had lost focus, being necessary to press shift + tabe 
  explorertxt
  and tabe to the list of files to be read again
  To reproduce the error, being within a folder in the file explorer, just paste new files or edit the names of these files, which after some operations NVDA no longer reads the list of files in Windows Explorer
  Attached a log that shows this fault

**Comment by fernando-jose-silva** [comment]
Thanks a lot, I noticed this problem in Windows 10, I remember that too in windwos 8.1 this also happened.

**Comment by ehollig** [comment]
@fernando-jose-silva, is this still an issue?

**Comment by fernando-jose-silva** [comment]
He was doing the tests now to answer the questioning.
Nvda is no longer dumb after copying or renaming files in explorer.
However I encountered a situation where nvda was muted when handling files in explorer.
Open an empty folder in explorer.
Copy to this folder at least 2 files.
select all files 2 in this folder and press shift + delete.
After confirming the deletion of all files press tab.
And now press 3 times the shift + tab.
The focus should be on the explorer tree list where there is the list of folders to navigate.
It is expected that the nvda will announce this list.
What happens is that the nvda is mute.
windows 10 17134.112 64bit
nvda next 15363
Follow the log:

IO - inputCore.InputManager.executeGesture (10:06:45.601):
Input: kb(laptop):shift+tab
IO - inputCore.InputManager.executeGesture (10:06:47.845):
Input: kb(laptop):shift+tab
IO - inputCore.InputManager.executeGesture (10:06:50.888):
Input: kb(laptop):shift+tab
IO - inputCore.InputManager.executeGesture (10:06:57.292):
Input: kb(laptop):NVDA+f1
DEBUGWARNING - watchdog._watcher (10:07:00.315):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 212, in <module>
  File "core.pyc", line 489, in main
  File "wx\core.pyc", line 2096, in MainLoop
  File "gui\__init__.pyc", line 959, in Notify
  File "core.pyc", line 461, in run
  File "queueHandler.pyc", line 83, in pumpAll
  File "queueHandler.pyc", line 50, in flushQueue
  File "scriptHandler.pyc", line 145, in _queueScriptCallback
  File "scriptHandler.pyc", line 187, in executeScript
  File "globalCommands.pyc", line 1438, in script_navigatorObject_devInfo
  File "logging\__init__.pyc", line 1174, in info
  File "logHandler.pyc", line 136, in _log
  File "gui\logViewer.pyc", line 113, in activate

INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (10:07:00.519):
Developer info for navigator object:
name: u'Nome'
role: ROLE_SPLITBUTTON
states: STATE_FOCUSABLE, STATE_READONLY, STATE_FOCUSED, STATE_COLLAPSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.UIA.UIColumnHeader object at 0x04B7FE10>
Python class mro: (<class 'NVDAObjects.UIA.UIColumnHeader'>, <class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: u'Classificado (Crescente)'
location: RectLTWH(left=337, top=239, width=340, height=32)
value: u''
appModule: <'explorer' (appName u'explorer', process ID 6764) at address 48896f0>
appModule.productName: u'Sistema operacional Microsoft\xae Windows\xae'
appModule.productVersion: u'10.0.17134.161'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 13895562
windowClassName: u'DirectUIHWND'
windowControlID: 0
windowStyle: 1442840576
windowThreadID: 11856
windowText: u''
displayText: u'Nome'
UIAElement: <POINTER(IUIAutomationElement) ptr=0xa3b1258 at 5616850>
UIA automationID: System.ItemNameDisplay
UIA frameworkID: DirectUI
UIA runtimeID: (6764, 573177344, 0)
UIA providerDescription: [pid:6764,providerId:0x0 Main(parent link):Unidentified Provider (unmanaged:explorerframe.dll)]
UIA className: UIColumnHeader
UIA patterns available: LegacyIAccessiblePattern, ExpandCollapsePattern, InvokePattern, ValuePattern

**Comment by ruifontes** [comment]
Yes, now it seens working alright...

Regarding the comment of Fernando José Silva, I confirm the error. 
Strangelly, Tab works and announce the new focus, but Shift+Tab does not 
work...

Tested also with Narrator and the result is the same...

Rui




Às 03:44 de 26/06/2018, Ethan Holliger escreveu:

**Comment by ehollig** [comment]
Closing as worksforme. As the second issue braught up by @fernando-jose-silva does not sound like an NVDA issue, I am still closing.

**Comment by fernando-jose-silva** [comment]
I believe the second problem is from nvda.

If in the rooms that I have passed do not make the sections referring to the deletion of the files, this wants to talk, open the explorer in a folder that contains files, press a tab, and then press 2 times the shift + tab and the nvda will announce the tree list perfectly.

 

De: Ethan Holliger <notifications@github.com> 
Enviada em: terça-feira, 26 de junho de 2018 13:43
Para: nvaccess/nvda <nvda@noreply.github.com>
Cc: fernando-jose-silva <fernandoj.silva21@gmail.com>; Mention <mention@noreply.github.com>
Assunto: Re: [nvaccess/nvda] NVDA does not read the file list in explorer after pasting a new file in the folder. (#5713)

 

Closing as worksforme. As the second issue braught up by @fernando-jose-silva <https://github.com/fernando-jose-silva>  does not sound like an NVDA issue, I am still closing.
